sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1262,18 +1257,20 @@ static void _init_from_device(gnrc_netif_t *netif)
     _update_l2addr_from_dev(netif);
 }
 
-static void _configure_netdev(netdev_t *dev)
+static void _check_netdev_capabilities(netdev_t *dev)
 {
-    /* Enable RX- and TX-complete interrupts */
-    static const netopt_enable_t enable = NETOPT_ENABLE;
-    int res = dev->driver->set(dev, NETOPT_RX_END_IRQ, &enable, sizeof(enable));
-    if (res < 0) {
-        DEBUG("gnrc_netif: enable NETOPT_RX_END_IRQ failed: %d\n", res);
-    }
-    if (IS_USED(MODULE_NETSTATS_L2) || IS_USED(MODULE_GNRC_NETIF_PKTQ)) {
-        res = dev->driver->set(dev, NETOPT_TX_END_IRQ, &enable, sizeof(enable));
-        if (res < 0) {
-            DEBUG("gnrc_netif: enable NETOPT_TX_END_IRQ failed: %d\n", res);
+    /* Check whether RX- and TX-complete interrupts are supported by the driver */
+    if (IS_ACTIVE(DEVELHELP)) {
+        netopt_enable_t enable = NETOPT_ENABLE;
+        int res = dev->driver->get(dev, NETOPT_RX_END_IRQ, &enable, sizeof(enable));
+        if ((res != sizeof(enable)) || (enable != NETOPT_ENABLE)) {
+            LOG_WARNING("NETOPT_RX_END_IRQ not implemented by driver\n");
```

Comment:
This new warning causes `tests_driver_xbee` to no longer build for `arduino-mega2560` due to increased RAM. (AVR is Harvard Arch, and even constant data is placed in RAM there.)

If we indeed want to keep `NETOPT_RX_END_IRQ`, it is IMO worth the have this warning. So far, only a hand full of drivers actually supports this netopt - but certainly every driver generates the corresponding event.